### PR TITLE
Added a new application "SimpleBackups" with 7 actions

### PR DESCRIPTION
**Pull request by sb-nour**
Added a new application "SimpleBackups"

- List Backups
- List Servers
- List Storages
- Get Download Link
- Pause Backup
- Resume Backup
- Run Backup

### CODE REVIEW

**Comment by pipedream-component-development** [comment]
Thank you so much for submitting this! We've added it to our backlog to review, and our team has been notified.

**Comment by pipedream-component-development** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:
- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

File: components/simplebackups/actions/get-download-link/get-download-link.mjs [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import app from "../../simplebackups.app.mjs";
+
+export default {
+  name: "Get Download Link",
+  version: "0.0.1",
+  key: "get-download-link",
```
Comment (luancazarine) [inline components/simplebackups/actions/get-download-link/get-download-link.mjs:6]:
```suggestion
  key: "simplebackups-get-download-link",
```

File: components/simplebackups/actions/get-download-link/get-download-link.mjs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  description: "Get the download link for a backup or backup log.",
```
Comment (luancazarine) [inline components/simplebackups/actions/get-download-link/get-download-link.mjs:7]:
```suggestion
  description: "Get the download link for a backup or backup log. [See the documentation](https://simplebackups.docs.apiary.io/#/reference/backups/get-download-link)",
```

File: components/simplebackups/actions/get-download-link/get-download-link.mjs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  props: {
+    app,
+    backupId: {
+      type: "integer",
+      label: "Backup ID",
+      description: "The ID of the backup to get the download link for.",
+      optional: false,
```
Comment (luancazarine) [inline components/simplebackups/actions/get-download-link/get-download-link.mjs:14]: This is not necessary as 'required' is a default option.
```suggestion
```

File: components/simplebackups/actions/list-storage/list-storage.mjs [hunk 4]
```diff
@@ -0,0 +1,49 @@
+import app from "../../simplebackups.app.mjs";
+
+export default {
+  name: "List Storages",
+  version: "0.0.1",
+  key: "list-storages",
+  description: "List storages.",
```
Comment (luancazarine) [inline components/simplebackups/actions/list-storage/list-storage.mjs:7]:
```suggestion
  description: "List all storage providers added to your SimpleBackups account. [See the documentation](https://simplebackups.docs.apiary.io/#/reference/storage/list-storage)",
```

File: components/simplebackups/actions/pause-backup/pause-backup.mjs [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import app from "../../simplebackups.app.mjs";
+
+export default {
+  name: "Pause Backup",
+  version: "0.0.1",
+  key: "pause-backup",
+  description: "Pause a backup.",
```
Comment (luancazarine) [inline components/simplebackups/actions/pause-backup/pause-backup.mjs:7]:
```suggestion
  description: "Pause a backup schedule given its ID. [See the documentation](https://simplebackups.docs.apiary.io/#/reference/backups/pause-backup)",
```

File: components/simplebackups/actions/pause-backup/pause-backup.mjs [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+  props: {
+    app,
+    backupId: {
+      type: "integer",
+      label: "Backup ID",
+      description: "The ID of the backup to get the download link for.",
+      optional: false,
```
Comment (luancazarine) [inline components/simplebackups/actions/pause-backup/pause-backup.mjs:14]:
```suggestion
```

File: components/simplebackups/actions/pause-backup/pause-backup.mjs [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    },
+  },
+  type: "action",
+  methods: {},
+  async run({ $ }) {
+    const { data } = await this.app.pauseBackup(this.backupId);
```
Comment (luancazarine) [inline components/simplebackups/actions/pause-backup/pause-backup.mjs:20]:
```suggestion
    const { data } = await this.app.pauseBackup($, this.backupId);
```

File: components/simplebackups/actions/resume-backup/resume-backup.mjs [hunk 8]
```diff
@@ -0,0 +1,42 @@
+import { axios } from "@pipedream/platform";
+
+export default {
+  name: "Resume Backup",
+  version: "0.0.1",
+  key: "resume-backup",
```
Comment (luancazarine) [inline components/simplebackups/actions/resume-backup/resume-backup.mjs:6]:
```suggestion
  key: "simplebackups-resume-backup",
```

File: components/simplebackups/actions/resume-backup/resume-backup.mjs [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+  description: "Resume a backup.",
```
Comment (luancazarine) [inline components/simplebackups/actions/resume-backup/resume-backup.mjs:7]:
```suggestion
  description: "Resume a backup schedule given its ID. [See the documentation](https://simplebackups.docs.apiary.io/#/reference/backups/resume-backup)",
```

File: components/simplebackups/actions/resume-backup/resume-backup.mjs [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+        "Content-Type": "application/json",
+        "Accept": "application/json",
+        "User-Agent": "@PipedreamHQ/pipedream v0.1",
+        "Authorization": `Bearer ${this.token}`,
+      },
+    });
+
+    console.log(response);
+
+    return response;
```
Comment (luancazarine) [inline components/simplebackups/actions/resume-backup/resume-backup.mjs:40]: You must to move it to app file and call it as was done in the other actions

File: components/simplebackups/actions/run-backup/run-backup.mjs [hunk 11]
```diff
@@ -0,0 +1,28 @@
+import app from "../../simplebackups.app.mjs";
+
+export default {
+  name: "Run Backup",
+  version: "0.0.1",
+  key: "run-backup",
```
Comment (luancazarine) [inline components/simplebackups/actions/run-backup/run-backup.mjs:6]:
```suggestion
  key: "simplebackups-run-backup",
```

File: components/simplebackups/actions/run-backup/run-backup.mjs [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+  description: "Run a backup.",
```
Comment (luancazarine) [inline components/simplebackups/actions/run-backup/run-backup.mjs:7]:
```suggestion
  description: "Run a backup given its ID. [See the documentation](https://simplebackups.docs.apiary.io/#/reference/backups/run-backup)",
```

File: components/simplebackups/actions/run-backup/run-backup.mjs [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+  props: {
+    app,
+    backupId: {
+      type: "integer",
+      label: "Backup ID",
+      description: "The ID of the backup to run.",
+      optional: false,
```
Comment (luancazarine) [inline components/simplebackups/actions/run-backup/run-backup.mjs:14]:
```suggestion
```

File: components/simplebackups/actions/run-backup/run-backup.mjs [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+    },
+  },
+  type: "action",
+  methods: {},
+  async run({ $ }) {
+    const { data } = await this.app.runBackup(this.backupId);
```
Comment (luancazarine) [inline components/simplebackups/actions/run-backup/run-backup.mjs:20]:
```suggestion
    const { data } = await this.app.runBackup($, this.backupId);
```

File: components/simplebackups/package.json [hunk 15]
```diff
@@ -0,0 +1,15 @@
+{
+  "name": "@pipedream/simplebackups",
+  "version": "0.0.1",
```
Comment (luancazarine) [inline components/simplebackups/package.json:3]:
```suggestion
  "version": "0.1.0",
```

File: components/simplebackups/package.json [hunk 16]
```diff
@@ -0,0 +1,15 @@
...
+  "description": "Pipedream SimpleBackups Components",
+  "main": "simplebackups.app.mjs",
+  "keywords": [
+    "pipedream",
+    "simplebackups"
+  ],
+  "author": "Pipedream <support@pipedream.com> (https://pipedream.com/)",
+  "license": "MIT",
```
Comment (luancazarine) [inline components/simplebackups/package.json:11]:
```suggestion
```

File: components/simplebackups/package.json [hunk 17]
```diff
@@ -0,0 +1,15 @@
...
+  "publishConfig": {
+    "access": "public"
+  }
```
Comment (luancazarine) [inline components/simplebackups/package.json:14]:
```suggestion
  },
  "dependencies": {
    "@pipedream/platform": "^1.5.1"
  }
```

File: components/simplebackups/simplebackups.app.mjs [hunk 18]
```diff
@@ -0,0 +1,157 @@
...
+    _apiUrl() {
+      return "https://my.simplebackups.io/api";
+    },
+
+    /**
+     * Logs the authentication keys to the console.
+     */
+    authKeys() {
+      console.log(Object.keys(this.$auth));
+    },
```
Comment (luancazarine) [inline components/simplebackups/simplebackups.app.mjs:41]:
```suggestion
```

File: components/simplebackups/simplebackups.app.mjs [hunk 19]
```diff
@@ -0,0 +1,157 @@
...
+      });
+    },
+
+    /**
+     * Retrieves the download link for a backup.
+     * @param {string} backupId - The ID of the backup.
+     * @param {string} [backupLogId=null] - The ID of the backup log.
+     * @returns {Promise} A promise that resolves with the download link.
+     */
+    async getDownloadLink(backupId, backupLogId = null) {
```
Comment (luancazarine) [inline components/simplebackups/simplebackups.app.mjs:80]:
```suggestion
    async getDownloadLink($, backupId, backupLogId = null) {
```

File: components/simplebackups/simplebackups.app.mjs [hunk 20]
```diff
@@ -0,0 +1,157 @@
...
+      let endpoint = `/backup/${backupId}/download-link`;
+      if (backupLogId) {
+        endpoint = `/backup/${backupId}/download-link/${backupLogId}`;
+      }
+      return this._makeAxiosRequest({
```
Comment (luancazarine) [inline components/simplebackups/simplebackups.app.mjs:85]:
```suggestion
      return this._makeAxiosRequest({
      $,
```

File: components/simplebackups/simplebackups.app.mjs [hunk 21]
```diff
@@ -0,0 +1,157 @@
...
+      });
+    },
+
+    /**
+     * Retrieves a list of storages.
+     * @param {string} [filters=""] - Additional filters for the list.
+     * @returns {Promise} A promise that resolves with the list of storages.
+     */
+    async listStorages(filters = "") {
+      return this._makeAxiosRequest({
```
Comment (luancazarine) [inline components/simplebackups/simplebackups.app.mjs:117]:
```suggestion
    async listStorages($, filters = "") {
      return this._makeAxiosRequest({
      $,
```

File: components/simplebackups/simplebackups.app.mjs [hunk 22]
```diff
@@ -0,0 +1,157 @@
...
+      });
+    },
+
+    /**
+     * Retrieves a list of backups.
+     * @param {string} [filters=""] - Additional filters for the list.
+     * @returns {Promise} A promise that resolves with the list of backups.
+     */
+    async listBackups(filters = "") {
+      return this._makeAxiosRequest({
```
Comment (luancazarine) [inline components/simplebackups/simplebackups.app.mjs:96]:
```suggestion
    async listBackups($, filters = "") {
      return this._makeAxiosRequest({
      $,
```

File: components/simplebackups/simplebackups.app.mjs [hunk 23]
```diff
@@ -0,0 +1,157 @@
...
+      });
+    },
+
+    /**
+     * Resumes a paused backup.
+     * @param {string} backupId - The ID of the backup to resume.
+     * @returns {Promise} A promise that resolves when the backup is resumed.
+     */
+    async resumeBackup(backupId) {
+      return this._makeAxiosRequest({
```
Comment (luancazarine) [inline components/simplebackups/simplebackups.app.mjs:140]:
```suggestion
    async resumeBackup($, backupId) {
      return this._makeAxiosRequest({
      $,
```

File: components/simplebackups/simplebackups.app.mjs [hunk 24]
```diff
@@ -0,0 +1,157 @@
...
+      });
+    },
+
+    /**
+     * Runs a backup.
+     * @param {string} backupId - The ID of the backup to run.
+     * @returns {Promise} A promise that resolves when the backup is run.
+     */
+    async runBackup(backupId) {
+      return this._makeAxiosRequest({
```
Comment (luancazarine) [inline components/simplebackups/simplebackups.app.mjs:152]:
```suggestion
    async runBackup($, backupId) {
      return this._makeAxiosRequest({
      $,
```

**Review by luancazarine — CHANGES_REQUESTED** [review]
Hi @sb-nour, thanks for your contribution! I just added some suggestions. Please let me know if you need some help.

File: components/simplebackups/actions/resume-backup/resume-backup.mjs (hunk 10 above)
Comment (sb-nour) [reply components/simplebackups/actions/resume-backup/resume-backup.mjs]: Thank you for letting me know, updated the action

**Comment by sb-nour** [comment]
> Hi @sb-nour, thanks for your contribution! I just added some suggestions. Please let me know if you need some help.

Thank you, I applied your suggestions. I appreciate your efforts!

**Review by luancazarine — APPROVED** [review]
LGTM!

**Comment by sergio-eliot-rodriguez** [comment]
Thanks for flagging @vunguyenhung , I'll talk a look

**Comment by sergio-eliot-rodriguez** [comment]
actually there's an app file in the registry for SimpleBackups, the app might be in "hidden" or "admin only" states. fyi @malexanderlim

**Review by luancazarine — APPROVED** [review]
LGTM!

**Pull request closed without merging**